core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1733,23 +1733,15 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
-  /**
-    * Convert an ACL binding filter to a Scala object.
-    * All ACL and resource fields must be specified (no UNKNOWN, ANY, or null fields are allowed.)
-    *
-    * @param filter     The binding filter as a Java object.
-    * @return           The binding filter as a scala object, or an exception if there was an error
-    *                   converting the Java object.
-    */
-  def toScala(filter: AclBindingFilter) : Try[(Resource, Acl)] = {
+  private def convertToResourceAndAcl(filter: AclBindingFilter): Try[(Resource, Acl)] = {
     filter.resourceFilter().resourceType() match {
       case AdminResourceType.UNKNOWN => return Failure(new InvalidRequestException("Invalid UNKNOWN resource type"))
       case AdminResourceType.ANY => return Failure(new InvalidRequestException("Invalid ANY resource type"))
-      case _ => {}
+      case _ =>
     }
     var resourceType: ResourceType = null
     try {
-      resourceType = ResourceType.fromString(filter.resourceFilter().resourceType().toString)
+      resourceType = ResourceType.fromString(filter.resourceFilter.resourceType.toString)
```

Comment:
It seems that we should move this logic to the `case _` above. We already have the `resourceType` variable in the case clause. And instead of try/catch, we can use the `Try` factory method that will catch and return a `Failure`.